products/views.py

```diff
@@ -27,4 +27,37 @@ def get(self, request, productgroup_id):
         ]
         mls    = [result['ml'] for result in results]
         prices = [result['price'] for result in results]
-        return JsonResponse({'products': results, 'mls': mls, 'prices': prices}, status=200)
\ No newline at end of file
+        return JsonResponse({'products': results, 'mls': mls, 'prices': prices}, status=200)
+
+
+class ProductListView(View):
+  def get(self,request):
+    try:
+      category_subcategory_id = request.GET.get('category_subcategory_id', None)
+      type_ml                 = request.GET.get('ml', None)
+      image                   = Image.objects.all()
+      
+      filter_set = {
+        'categorysubcategory_id': category_subcategory_id,
+        'ml'                    : type_ml
+      }
+      
+      products = Product.objects.filter(**filter_set) 
+
+      products = [{   
+        'id'             : product.id,
+        'name'           : product.name,
+        'ml'             : product.ml,
+        'price'          : product.price,
+        'productgroup_id': product.productgroup.id,
+        'image'          : [img for img in image.filter(product_id = product.id).values('image_url')],
+        'subcategory': {
+          'subcategory_id'  : product.categorysubcategory.subcategory.id,
+          'subcategory_name': product.categorysubcategory.subcategory.name
+        }
+      } for product in products]
+
+      return JsonResponse({'products': products}, status= 200)
+
+    except:
+      return JsonResponse({'message': 'KEY_ERROR'}, status=400) 
```

Comment:
명확하게 error를 핸들링해주셔야 합니다. 단순히 except만 사용하는 경우는 피해주세요~
```suggestion
    except KeyError:
      return JsonResponse({'message': 'KEY_ERROR'}, status=400) 
```